test/capped_sto.js

```diff
@@ -665,22 +673,149 @@ contract('CappedSTO', accounts => {
+            );
+            assert.equal(
+                await I_CappedSTO_Array_ETH[1].endTime.call(),
+                endTime_ETH2,
+                "STO Configuration doesn't set as expected"
+            );
+            assert.equal(
+                (await I_CappedSTO_Array_ETH[1].cap.call()).toNumber(),
+                cap,
+                "STO Configuration doesn't set as expected"
+            );
+            assert.equal(
+                await I_CappedSTO_Array_ETH[1].rate.call(),
+                rate,
+                "STO Configuration doesn't set as expected"
+            );
+            assert.equal(
+                await I_CappedSTO_Array_ETH[1].fundraiseType.call(),
+                fundRaiseType,
+                "STO Configuration doesn't set as expected"
+            );
+        });
+
+        it("Should successfully invest in second STO", async() => {
+
+            balanceOfReceiver = await web3.eth.getBalance(account_fundsReceiver);
+
+            let tx = await I_GeneralTransferManager.modifyWhitelist(
+                account_investor3,
+                fromTime,
+                toTime,
+                expiryTime,
+                true,
+                {
+                    from: account_issuer,
+                    gas: 500000
+                });
+
+            assert.equal(tx.logs[0].args._investor, account_investor3, "Failed in adding the investor in whitelist");
+
+            // Jump time to beyond STO start
+            await increaseTime(duration.days(2));
+
+            await I_CappedSTO_Array_ETH[1].buyTokens(account_investor3, { from : account_investor3, value: web3.utils.toWei('1', 'ether') });
+
+            assert.equal(
+                (await I_CappedSTO_Array_ETH[1].fundsRaised.call())
+                .dividedBy(new BigNumber(10).pow(18))
+                .toNumber(),
+                1
+            );
+
+            assert.equal(await I_CappedSTO_Array_ETH[1].getNumberInvestors.call(), 1);
+
+            assert.equal(
+                (await I_SecurityToken_ETH.balanceOf(account_investor3))
+                .dividedBy(new BigNumber(10).pow(18))
+                .toNumber(),
+                1000
+            );
+        });
+    });
+
+    describe("Test cases for reaching limit number of STO modules", async() => {
+
+        it("Should successfully attach STO modules up to the limit and fail at the limit", async () => {
+            let startTime = latestTime() + duration.days(1);
+            let endTime = startTime + duration.days(30);
+
+            await I_PolyToken.getTokens(cappedSTOSetupCost*19, token_owner);
+            await I_PolyToken.transfer(I_SecurityToken_ETH.address, cappedSTOSetupCost*19, { from: token_owner});
+            let bytesSTO = web3.eth.abi.encodeFunctionCall(functionSignature, [startTime, endTime, cap, rate, fundRaiseType, account_fundsReceiver]);
+
+            for (var STOIndex = 2; STOIndex < 20; STOIndex++) {
```

Comment:
SecurityToken has this var: `uint8 public constant MAX_MODULES = 20;`
instead of hardcoding 20 in the for loop use that one since MAX_MODULES might change in the future.